Shopify/ruby-lsp
# Add show syntax tree request

### Motivation

Add a show syntax tree request, so that developers can easily inspect the AST of a given file. This is helpful when trying to figure out how Ruby understands code.

### Implementation

The new custom request just returns an object containing the pretty printed AST, so that we can put it in a file and display to the user.

In the future, we can also support
- Showing the AST for a selection
- Using some webview visualization in VS Code to allow navigating the AST alongside the file

### Automated Tests

Added a test for it.

### Manual Tests

1. Start the LSP on this branch
2. Open a Ruby file
3. CMD + SHIFT + P: `Ruby LSP: show syntax tree`
4. Verify the syntax tree is displayed properly
